GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -124,39 +124,112 @@ public static bool TrySetRequiredGitConfigSettings(Enlistment enlistment)
             //
             Dictionary<string, string> requiredSettings = new Dictionary<string, string>
             {
+                 // When running 'git am' it will remove the CRs from the patch file by default. This causes the patch to fail to apply because the
+                 // file that is getting the patch applied will still have the CRs. There is a --keep-cr option that you can pass the 'git am' command
+                 // but since we always want to keep CRs it is better to just set the config setting to always keep them so the user doesn't have to
+                 // remember to pass the flag. https://mseng.visualstudio.com/VSOnline/_git/GVFS/pullrequest/244891?_a=overview.
                 { "am.keepcr", "true" },
+
+                // Update git settings to enable optimizations in git 2.20
+                // Set 'checkout.optimizeNewBranch=true' to enable optimized 'checkout -b'
                 { "checkout.optimizenewbranch", "true" },
+
+                // Needed to prevent line ending conversion. Prevents git from auto-replacing line endings with Crlf.
                 { "core.autocrlf", "false" },
+
+                // commit graph. https://dev.azure.com/mseng/AzureDevOps/_git/GVFS/pullrequest/299081
                 { "core.commitGraph", "true" },
+
+                // Perf - Git for Windows uses this to bulk-read and cache lstat data of entire directories (instead of doing lstat file by file).
                 { "core.fscache", "true" },
+
+                // Turns on all special gvfs logic. https://github.com/microsoft/scalar/blob/935c50110b31c440aae8e1cdc4ffea72bc3be01f/Scalar/CommandLine/ScalarVerb.cs#L35
                 { "core.gvfs", "true" },
+
+                // Use 'multi-pack-index' builtin instead of 'midx' to match upstream implementation
                 { "core.multiPackIndex", "true" },
+
+                // Perf - Enable parallel index preload for operations like git diff
                 { "core.preloadIndex", "true" },
                 { "core.safecrlf", "false" },
+
+                // Possibly cause hydration while creating untrackedCache.
                 { "core.untrackedCache", "false" },
                 { "core.repositoryformatversion", "0" },

```

Comment:
This is the discussion from when `core.repositoryformatversion` was added (https://dev.azure.com/mseng/AzureDevOps/_git/GVFS/pullrequest/235679):


Saeed Noursalehi
7/10/2017
Why are we calling 'git init'? Why not just create a new empty config file?


Saeed Noursalehi
7/10/2017
It looks like that's what you're doing down below. I don't understand what the 'git init' is for.


Forest Willard
7/10/2017
git init populates four entries that GVFS does not. I don't know much about them, but I figured this is how clone initializes the config, so it was best to keep it in line with that.

[core]
repositoryformatversion = 0
filemode = false
bare = false
logallrefupdates = true


Saeed Noursalehi
7/11/2017
It was just surprising to see "git init" called on an existing repo. What other side effects does it have besides recreating the config file?

If it's just a matter of creating those 4 config entries, I'd be fine with adding them to the list of entries in GVFSVerb, since they all seem like sensible values, and we're currently relying on them anyway so might as well make it explicit.


Forest Willard
7/11/2017
From the git man:
Running git init in an existing repository is safe. It will not overwrite things that are already there. The primary reason for rerunning git init is to pick up newly added templates (or to move the repository to another place if --separate-git-dir is given).

So really, the primary side effect would be "accidentally fixing" any missing folders in .git (refs/heads, objects, etc). but I'm not opposed to just adding those entries to simplify/clarify things here.

Reply:
Added - `This is to match what git init does.`